addon/chart-data-updater.js

```diff
@@ -15,12 +15,39 @@ export default Ember.Object.extend({
     var chart = this.get('chart');
     var self = this;
 
+	
+	//If new data has less columns
+    if (chart.datasets.length > datasets.length) {
+      var temp = Array();
+      chart.datasets.forEach(function(dataset, i) {
+		  //copy over only that amount
+        if (datasets[i]) {
+          temp.push(chart.datasets[i]);
+        }
+      });
+	  //and place into the chart data
+      chart.datasets = temp;
+    }
+
+	//If new data has more colmns
```

Comment:
You are right here except it still does work because of the error catching that you have in place
<code>
} catch (e) {
          if (e instanceof TypeError) {
            self.set('redraw', true);
          } else {
</code>

The data in the chart data set is not what is expected and therefore the redraw flag is set to true and then finally (as discussed above )
<code>  return this.get('redraw'); <.code> 
is called and the new data set is used to produce a new chart.

My changes are messy and I will happily clean it up a bit later. For example, really it would just be better to copy the new data set into the chart and return with a redraw flag set to true as soon as the column size difference is detected. This stops the rest of the code getting executed unnecessarily.